File: LLM-Translations/GPT-4/avatar/Python/C/codeforces_30_A.c
#include<stdio.h>
#include<math.h>

int task4(int A, int B, int n) {
    int X;
    for(X = -1000; X <= 1000; X++) {
        if(A * pow(X, n) == B) {
            return X;
        }
    }
    return -1; // Returning -1 to indicate "No solution"
}

int main() {
    int A, B, n, result;
    scanf("%d %d %d", &A, &B, &n);
    result = task4(A, B, n);
    if(result != -1) {
        printf("%d", result);
    } else {
        printf("No solution");
    }
    return 0;
} // End of Code

File: LLM-Translations/GPT-4/avatar/Python/Java/codeforces_30_A.java
public class codeforces_30_A {
    public static void main(String[] args) {
        int A = Integer.parseInt(args[0]);
        int B = Integer.parseInt(args[1]);
        int n = Integer.parseInt(args[2]);
        System.out.println(task4(A, B, n));
    }

    public static String task4(int A, int B, int n) {
        for (int X = -1000; X <= 1000; X++) {
            if (A * Math.pow(X, n) == B) {
                return Integer.toString(X);
            }
        }
        return "No solution";
    }
}
// End of Code
